gsy_framework/sim_results/kpi.py

```diff
@@ -151,49 +153,41 @@ def update_area_kpi(self, area_dict: Dict, core_stats: Dict):
         self.consumer_ess_set = set()  # keeps set of house's consuming/ess devices
         self.fit_revenue = 0.  # revenue achieved by producing selling energy via FIT scheme
         self.utility_bill = 0.  # cost of energy purchase from energy supplier
-        self.base_case_cost = 0.  # standard cost of a house participating in FIT scheme
         self.gsy_e_cost = 0.  # standard cost of a house participating in GSy Exchange
-        self.saving_absolute = 0.  # savings achieved by a house via participating in D3A
-        self.saving_percentage = 0.  # savings in percentage wrt FIT via participating in D3A
 
-    def calculate_savings_kpi(self, area_dict: dict, core_stats: dict, gf_alp: float):
-        """Calculates the referenced saving from feed-in tariff based participation vs D3A
+    def calculate_savings_kpi(self, area_dict: Dict, core_stats: Dict, grid_fee_along_path: float):
+        """Calculates the referenced saving from feed-in tariff based participation vs GSY-E
         Args:
             area_dict: contain nested area info
             core_stats: contain area's raw/key statistics
-            gf_alp: grid_fee_along_the_path - cumulative grid fee from root to target area
+            grid_fee_along_path: grid_fee_along_the_path -
+            cumulative grid fee from root to target area
 
         """
-        self.populate_consumer_producer_sets(area_dict)
-
-        # fir_excl_gf_alp: feed-in tariff rate excluding grid fee along path
-        fir_excl_gf_alp = self.get_feed_in_tariff_rate_excluding_path_grid_fees(
-            core_stats.get(area_dict["uuid"], {}), gf_alp)
-        # mmr_incl_gf_alp: market maker rate include grid fee along path
-        mmr_incl_gf_alp = self.get_market_maker_rate_including_path_grid_fees(
-            core_stats.get(area_dict["uuid"], {}), gf_alp)
+        self._populate_consumer_producer_sets(area_dict)
+
+        feed_in_tariff = self._get_feed_in_tariff_rate_excluding_path_grid_fees(
+            core_stats.get(area_dict["uuid"], {}), grid_fee_along_path)
+        market_maker_rate = self._get_market_maker_rate_including_path_grid_fees(
+            core_stats.get(area_dict["uuid"], {}), grid_fee_along_path)
+
         for trade in core_stats.get(area_dict["uuid"], {}).get("trades", []):
-            if (trade["seller_origin_id"] in self.producer_ess_set and
-                    trade["buyer_origin_id"] not in self.consumer_ess_set):
-                self.fit_revenue += fir_excl_gf_alp * trade["energy"]
-                self.gsy_e_cost -= trade["price"]
-            if trade["buyer_origin_id"] in self.consumer_ess_set and \
-                    trade["seller_origin_id"] not in self.producer_ess_set:
-                self.utility_bill += mmr_incl_gf_alp * trade["energy"]
+            if trade["seller_origin_id"] in self.producer_ess_set:
+                if feed_in_tariff > 0:
+                    self.fit_revenue += feed_in_tariff * trade["energy"]
```

Comment:
The fit_revenue should only be calculated if the feed_in_tariff is larger than the grid_fee a long the path.